degica/barcelona-cli
# Fix web container port set to zero if not specified

# Context

A bug was introduced a while ago with 0.9.3 prerelease where if `web_container_port` was not set in `barcelona.yml` it would send 0 to barcelona. This PR fixes it.

# How to test
You don't need a working barcelona server for testing this. Just checking the debug output is enough.
1. Check out https://github.com/degica/barcelona-hello
2. Open the `barcelona.yml` and delete the line that says `web_container_port: 8080`
3. Go into the checked out directory and run `./bcn -d deploy -e hello`

You should see output that looks something like:

```
PATCH /v1/heritages/hello?debug=true HTTP/1.1
Host: barcelona.localhost.labs.degica.com
User-Agent: Go-http-client/1.1
Content-Length: 293
Accept: application/json
Content-Type: application/json
X-Barcelona-Token: [filtered]
Accept-Encoding: gzip

{"name":"hello","image_name":"quay.io/dsiawdegica/hello","before_deploy":null,"scheduled_tasks":[],"services":[{"name":"web","memory":256,"command":"sh /start.sh","service_type":"web","force_ssl":false,"hosts":[],"listeners":[{"endpoint":"hello","health_check_path":"/"}]}],"environment":null}
^C
```

Note that the JSON shouldn't have any `web_container_port` param in it.

4. Put the parameter back
5. Run `./bcn -d deploy -e hello`

Check that the `"web_container_port": 8080` is now present in the JSON
